gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -615,6 +615,8 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
       }
       if (PropertiesUtils.getPropAsBoolean(jobConfig, ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false")) {
         _log.info("RunImmediately requested, hence executing FlowSpec: " + addedSpec);
+        // Use 0 for trigger event time of a run-immediately flow
+        jobConfig.setProperty(ConfigurationKeys.ORCHESTRATOR_TRIGGER_EVENT_TIME_MILLIS_KEY, "0");
```

Comment:
Is there a scenario where we don't set this key? Wouldn't it be better to handle this on the reader side e.g. the `orchestrator` class?
I see this snippet:
```
          _log.warn("Skipping execution of spec: {} because missing trigger timestamp in job properties",
              jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY));
          flowMetadata.put(TimingEvent.METADATA_MESSAGE, "Flow orchestration skipped because no trigger timestamp "
              + "associated with flow action.");
          if (this.eventSubmitter.isPresent()) {
            new TimingEvent(this.eventSubmitter.get(), TimingEvent.FlowTimings.FLOW_FAILED).stop(flowMetadata);
          }
          return;
        }
 ```

Reply:
Good point, I don't envision a scenario we would rather skip flows rather then use the default trigger timestamp of 0 and execute them. Removed this check and updated the PR